job-service-db/src/main/resources/procedures/taskCollapse.sql

```diff
@@ -0,0 +1,127 @@
+CREATE OR REPLACE FUNCTION task_collapse(tasks VARCHAR[])
```

Comment:
copyright ok
unit tests added 
perf test: 

   - Inserted some lines in the completed_subtask-report table
   - Tested without the function, then with it

```
For 65318 lines Before  2541.163 ms / After 61631.399 ms
```

It slows down the process

`

Reply:
Actually, I am unsure if the task list passed in the test is really relevant. I will try to do it again with the list generated from one of the last tests. Also, the algorithm can be improved for sure.

For example, instead of checking all elements one by one when a star appears, we could try to "predict". 
If we have 4* for tasks[1] then we check if tasks[4] has the same body and ends with 1. 

If so, we collapse. Otherwise, we skip to the next "*"